burn-tensor/src/tests/ops/abs.rs

```diff
@@ -0,0 +1,16 @@
+#[burn_tensor_testgen::testgen(abs)]
+mod tests {
+    use super::*;
+    use burn_tensor::{Data, Tensor};
+
+    #[test]
+    fn should_support_abs_ops() {
+        let data = Data::from([[0.0, -1.0, 2.0], [3.0, 4.0, -5.0]]);
+        let tensor = Tensor::<TestBackend, 2>::from_data(data);
+
+        let data_actual = tensor.abs().into_data();
+
+        let data_expected = Data::from([[0.0, 1.0, 2.0], [3.0, 4.0, 5.0]]);
+        data_expected.assert_approx_eq(&data_actual, 3);
+    }
```

Comment:
We also need tests for Int tensor. Please see this example: https://github.com/burn-rs/burn/pull/550/files#diff-0479313033166bd3d20e80b633111dada0400a368a82e66c96754edade7268dcR17-R22

Reply:
In the example you have submitted, why is there no tensor.rs implementation for burn-autodiff?

Reply:
Also, there is one more thing I do not quite understand.
In burn-tensor/src/tensor/ops/int_tensor.rs you provide implementation for int_clamp_min and int_clamp_max.
However, in the test (burn-tensor/src/tests/ops/clamp.rs) for integer implementation you use clamp_min and clamp_max methods. Why is that? 
I imagine that there is something going on in between burn-tensor/src/tensor/api/ and  burn-tensor/src/tensor/ops/, but I cannot really see that.

Reply:
> In the example you have submitted, why is there no tensor.rs implementation for burn-autodiff?

Good point. Initially I had it. But in my offline discussion with @nathanielsimard during the review, he said since I have implemented the default method using the existing ops, my backward pass implementation resembled what autodif would have derived, it's better not implement it for Auto diff backend. He said you only need it if there is no default and/or you want a more optimized version. 

In your case, you do not have the default implementation ( ^2.sqrt() would be default but it's slow), you'll need to implement the backward pass.

Reply:
> Also, there is one more thing I do not quite understand. In burn-tensor/src/tensor/ops/int_tensor.rs you provide implementation for int_clamp_min and int_clamp_max. However, in the test (burn-tensor/src/tests/ops/clamp.rs) for integer implementation you use clamp_min and clamp_max methods. Why is that? I imagine that there is something going on in between burn-tensor/src/tensor/api/ and burn-tensor/src/tensor/ops/, but I cannot really see that.

Yes, it's confusing and that's another reason we need a guide/documentation. 

Method names with `int_` come from the backend implementation. I think the prefix is to avoid collision. The test that I added actually tests the official Tensor API using a backend.

Reply:
I have added integer implementation, but I have some doubts. I would be grateful if you could have a look